### PR TITLE
173 finalize kni production for 1020

### LINKED ISSUE #173: Finalize KNI production for 10.2.0
We have to finalize KNI for the current release:
- packaging on windows: a ZIP file
- update the KNI-tutorial project: 
    - add installation notes for windows, debian and centos
    - add download link for all KNI release

### PR DESCRIPTION
**Pull request by bruno-at-orange**
- Add the KNI.zip production in pack-nsis
- update documentation in template-README.md

The package kni.zip is produced and seems OK.
For test purpose I have manually triggered the KNI tutorial generation [tag 10.2.0](https://github.com/KhiopsML/KNI-tutorial/tree/10.2.0)
But I have not tested the upload of KNI in the release (and then links are broken in the tutorial). We have to decide when to trigger the tutorial generation.

### CODE REVIEW

File: .github/workflows/pack-nsis.yml [hunk 1]
```diff
@@ -82,11 +82,20 @@ jobs:
...
+        shell: bash
+        run: |-
+          cd build/windows-msvc-release && CPACK -G ZIP 
       - name: Upload the Installer Artifact
         uses: actions/upload-artifact@v4
         with:
           name: khiops-setup
           path: ./packaging/windows/nsis/khiops-${{ env.KHIOPS_VERSION }}-setup.exe
+      - name: Upload the Installer Artifact
+        uses: actions/upload-artifact@v4
```
Comment (folmos-at-orange) [inline .github/workflows/pack-nsis.yml]: I'd put `Upload the KNI zip`

File: .github/workflows/pack-nsis.yml [hunk 2]
```diff
@@ -142,3 +155,13 @@ jobs:
...
+      - name: Upload KNI zip to the release
+        uses: ncipollo/release-action@v1.14.0
+        with:
+          allowUpdates: true
+          artifacts: KNI-*.zip
+          body: '**For testing purposes only**'
+          draft: false
+          makeLatest: false
+          prerelease: true
+          updateOnlyUnreleased: true
```
Comment (folmos-at-orange) [inline .github/workflows/pack-nsis.yml]: I think you can use just one release action with:

`artifacts: "khiops-*-setup.exe,KNI-*.zip"`

As specified in the [action's docs](https://github.com/ncipollo/release-action?tab=readme-ov-file#example)

**Review by folmos-at-orange — CHANGES_REQUESTED** [review]
Small stuff.

I'd test that creating a tag creates the expected release

```bash
# On your local branch 173-finalize-kni-production-for-1020
git tag 10.2.1-a1
git push --tags
```
Then you can erase the release and the tags.

File: .github/workflows/pack-nsis.yml (hunk 1 above)
Comment (bruno-at-orange) [reply .github/workflows/pack-nsis.yml]: done

File: .github/workflows/pack-nsis.yml (hunk 2 above)
Comment (bruno-at-orange) [reply .github/workflows/pack-nsis.yml]: done

**Comment by folmos-at-orange** [comment]
I did a temporary PR with my suggestions for the readmes. I think this is easier to do than comments for non-code changes. Make the changes you think are ok and then i'll remove the PR and branch later.

https://github.com/KhiopsML/khiops/pull/176/files

**Comment by bruno-at-orange** [comment]
I have added the java test on windows and linux. (by the way, the action nvoxland/jar-download-action works only on Linux, I replaced it with a simple curl).
I have taken into account your suggestion on the readme files (and a few from Marc)

**Comment by bruno-at-orange** [comment]
I have added curl to all Dockerfile and removed wget (it was use only to download cmake).
I have removed the custom installation of cmake in rocky8 because cmake is now up to date

File: packaging/dockerfiles/Dockerfile.rocky9 [hunk 3]
```diff
@@ -9,19 +9,17 @@ RUN true \
...
+  rpm-build \
   && dnf clean all \
   && rm -rf /var/cache/dnf/ \
   # Note: We install cmake from kitware's site to have a more recent version \
   # If rocky9 eventually supports cmake >= 3.22 we may use dnf again \
-  && wget "https://github.com/Kitware/CMake/releases/download/v3.26.5/cmake-3.26.5-linux-x86_64.sh" \
+  && curl -LO "https://github.com/Kitware/CMake/releases/download/v3.26.5/cmake-3.26.5-linux-x86_64.sh" \
   && sh cmake-3.26.5-linux-x86_64.sh --prefix=/usr/local/ --exclude-subdir \
   && rm -f cmake-3.26.5-linux-x86_64.sh \
-  && dnf remove -y wget "dnf-command(config-manager)"\
```
Comment (folmos-at-orange) [inline packaging/dockerfiles/Dockerfile.rocky9:26]: I think we should keep the removal of `dnf-command(config-manager)` which is needed only to install `ninja-build`.

File: packaging/dockerfiles/Dockerfile.rocky8 [hunk 4]
```diff
@@ -9,20 +9,14 @@ RUN true \
...
+  rpm-build \
+  curl \
   && dnf clean all \
   && rm -rf /var/cache/dnf/ \
-  # Note: We install cmake from kitware's site to have a more recent version \
-  # If rocky8 eventually supports cmake >= 3.22 we may use dnf again \
-  && wget "https://github.com/Kitware/CMake/releases/download/v3.26.5/cmake-3.26.5-linux-x86_64.sh" \
-  && sh cmake-3.26.5-linux-x86_64.sh --prefix=/usr/local/ --exclude-subdir \
-  && rm -f cmake-3.26.5-linux-x86_64.sh \
-  && dnf remove -y wget "dnf-command(config-manager)" \
```
Comment (folmos-at-orange) [inline packaging/dockerfiles/Dockerfile.rocky8:27]: I think we should leave the custom CMake download and reinstate the removal of `dnf-command(config-manager)` which is used only to install `ninja-build`.

**Review by folmos-at-orange — CHANGES_REQUESTED** [review]
Small details on the dockerfiles

File: packaging/dockerfiles/Dockerfile.rocky8 (hunk 4 above)
Comment (bruno-at-orange) [reply packaging/dockerfiles/Dockerfile.rocky8:27]: my mistake for the dnf-command
but why should we keep a non standard cmake installation: the dnf one is perfectly fine. We can fix the cmake version if you like...
In the comment we have: "f rocky8 eventually supports cmake >= 3.22 we may use dnf again"

Comment (folmos-at-orange) [reply packaging/dockerfiles/Dockerfile.rocky8:27]: Ahh, after seeing the logs it seem they have updated cmake to 3.26 in their repos. Then I'm totally ok to remove the custom installer.

Comment (bruno-at-orange) [reply packaging/dockerfiles/Dockerfile.rocky8:27]: OK, I just added the removal of dnf-command

**Review by folmos-at-orange — APPROVED** [review]
LGTM

**Pull request merged**